### PR TITLE
Add word validation when generating passphrase

### PR DESCRIPTION
**Pull request by Nasicus**
I always liked the word validation which we have in ark-desktop, so I added the same validation here.

A few notes:
- I designed the word validation so that it will also work with a different nubmer of words, or more words, etc. => we could also make it random, each time a passphrase is generated (so the user cannot simply remember the 3, 6 and 9 word)
- I added a little gimmick, that when we are on `devnet` the words area already prefilled, so we can go faster in testing (for this purpose I added a gett er `isTestNet` to the `UserDataProvider` so we could also use this feature elsewhere (if you approve it of course ;) )

By the way if you omitted this feature on purpose on mobile, you can of course decline the PR, I really just wanted to play a little with cordova and get into it and find this a cool feature :)

Also: How do we handle translations? I'm not fluent in swedish (=I do not speak a single word of it), so I omitted it.

Tested on:
 - Browser
- Android

screenshot_20171231-112647

### CODE REVIEW

File: src/providers/user-data/user-data.ts [hunk 1]
```diff
@@ -30,6 +30,10 @@ export class UserDataProvider {
   public onUpdateWallet$: Subject<Wallet> = new Subject();
   public onSelectProfile$: Subject<Profile> = new Subject();
 
+  public get isTestNet(): boolean {
```
Comment (alexbarnsley) [inline src/providers/user-data/user-data.ts]: A few things regarding this. I would calling it "isDevnet". You can also tweak the code to match in the mainnet-only directive (and change that code to call `isDevnet` too. From the directive:

```
    if (this.userDataProvider.currentNetwork.type === NetworkType.Devnet) {
      this.elementRef.nativeElement.style.display = 'none';
    }
```

No need for the constant once you use this

**Comment by alexbarnsley** [comment]
It looks good! I've not tested yet. One tweak that I can see I think but no functionality currently. I don't see why we wouldn't leave this out. There are plans to update the process on the desktop wallet, so it may be the same here. Take a look at my comment then I'll give it a test. Thinking about it, I think it should be 3 random words.

**Comment by Nasicus** [comment]
Ok done.

Note that the random algorithm was a little tricky, or rather I made it that.
It occured sometimes that the random words, were words following next to eachother, e.g.:  1,23 or 5,6,7 or something like that.
So I basically added a check to the method which verifies, that there's always at least one word between the next one.
Don't know if this is an overkill or overcomplicated, but I find it better that way.

**Comment by alexbarnsley** [comment]
Interestingly my initial test showed 3, 6, 9 as my word options 😂  turned out it was a big coincidence. However, I really like it! I have 1 request: Could it only validate the words when the button is pressed? It might not be necessary. What do you think? 

There's an issue with scrolling when I select the bottom field on Android, but it's the same when I go to the Send screen and select SmartBridge. Any ideas @luciorubeens?

**Comment by alexbarnsley** [comment]
If you could just fix the conflict, also, that would be beautiful 😄

**Comment by Nasicus** [comment]
Why dont you like the on type validating? I added it because I thought its better than waiting until "Next" is pressed. But I can obviously change it if you want.

**Comment by alexbarnsley** [comment]
I like it from a UX perspective. I think I was thinking along the lines of brute force. I don't think it applies here so leave it :)

**Comment by alexbarnsley** [comment]
Just fix the conflicts and I'll merge 😁

**Comment by alexbarnsley** [comment]
@Nasicus t-12 hours :trollface:

**Comment by Nasicus** [comment]
@alexbarnsley 
T+17 hours :trollface: 
Done.

**Comment by alexbarnsley** [comment]
Tested on Mainnet and Devnet. Can't fault it. All working as it should be. Nice work! 👌

**Pull request merged**